src/main/scala/models/Opportunity.scala

```diff
@@ -26,7 +26,21 @@ case class Opportunity(
                         duplicatedFrom: Option[OpportunityId],
                         description: Seq[OpportunityDescriptionSection]
                       ) {
+
   lazy val summary: OpportunitySummary = OpportunitySummary(id, title, startDate, endDate, value, publishedAt, duplicatedFrom)
+
+  lazy val statusString: String = {
+    publishedAt.isDefined match {
+      case true if isEndDatePassed => "Closed"
+      case true if isStartDatePassed => "Open"
+      case true => "Queued"
+      case false => "Draft"
+    }
+  }
+
+  lazy val isStartDatePassed = startDate.isEqual(LocalDate.now) || startDate.isBefore(LocalDate.now)
```

Comment:
I've done a little tidy up but otherwise looks good. I thought about introducing a parameter to the function to provide a reference date to avoid the issue that Jon highlighted, but in fact it's not such a big issue that we need to do that now.